devicetypes/qubino/qubino-flush-2-relay.src/qubino-flush-2-relay.groovy

```diff
@@ -0,0 +1,451 @@
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		try {
+			if (state.currentPreferencesState."$it.key".status == "syncPending") {
+				commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+				commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+			} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+				commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+				commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+			}
+		} catch (e) {
+			log.warn "There's been an issue with preference: ${it.key}"
+		}
+	}
+	sendHubCommand(commands)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd, ep = null) {
+	// Preferences template begin
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find( {it.parameterNumber == cmd.parameterNumber} )
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+	// Preferences template end
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private isPreferenceChanged(preference) {
+	if (settings."$preference.key" != null) {
+		return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+	} else {
+		return false
+	}
+}
+
+def parse(String description) {
+	def result = null
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText:description, isStateChange:true)
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd, null)
```

Comment:
nit, but if you're setting this as a default argument in your signatures, you shouldn't need to pass `null`